zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStore.scala

```diff
@@ -277,71 +220,48 @@ class CassieSpanStore(
    * API Implementation
    */
   override def close(deadline: Time): Future[Unit] =
-    FuturePool.unboundedPool { keyspace.close() }
+    FuturePool.unboundedPool { repository.close() }
 
-  // TODO: break these into smaller batches?
   def apply(spans: Seq[Span]): Future[Unit] = {
     SpansStoredCounter.incr(spans.size)
 
-    val traces = Traces.batch()
-    val serviceNames = ServiceNames.batch()
-    val spanNames = SpanNames.batch()
-    val serviceNameIdx = ServiceNameIndex.batch()
-    val serviceSpanNameIdx = ServiceSpanNameIndex.batch()
-    val annotationsIdx = AnnotationsIndex.batch()
-    val durationIdx = DurationIndex.batch()
-
     spans foreach { span =>
-      traces.insert(span.traceId, Column[String, ThriftSpan](createSpanColumnName(span), span.toThrift, None, Some(spanTtl)))
+      repository.storeSpan(span.traceId, createSpanColumnName(span), spanCodec.encode(span.toThrift), spanTtl.inSeconds)
       if (shouldIndex(span)) {
         SpansIndexedCounter.incr()
-        indexServiceName(serviceNames, span)
-        indexSpanNameByService(spanNames, span)
-        indexTraceIdByName(serviceNameIdx, serviceSpanNameIdx, span)
-        indexByAnnotations(annotationsIdx, span)
-        indexSpanDuration(durationIdx, span)
+        indexServiceName(span)
+        indexSpanNameByService(span)
+        indexTraceIdByName(span)
+        indexByAnnotations(span)
+        indexSpanDuration(span)
       }
     }
 
-    Future.collect(Seq(
-      traces,
-      serviceNames,
-      spanNames,
-      serviceNameIdx,
-      serviceSpanNameIdx,
-      annotationsIdx,
-      durationIdx
-    ).map(_.execute())).unit
+    Future.Unit
   }
 
   def setTimeToLive(traceId: Long, ttl: Duration): Future[Unit] = {
-    Traces.getRow(traceId) flatMap { row =>
-      val traces = Traces.batch()
-      row.values.asScala foreach { col =>
-        traces.insert(traceId, col.copy(timestamp = None, ttl = Some(ttl)))
-      }
-      traces.execute().unit
+    getSpansByTraceId(traceId).get foreach { span =>
```

Comment:
Ah, it's a Future. You'll have to use

```scala
getSpansByTraceId(traceId) map { spans => spans foreach { span =>
```
Then you won't need to return Future.Unit in the last line.